include/nlohmann/detail/exceptions.hpp

```diff
@@ -131,16 +131,30 @@ class exception : public std::exception
         {
             return concat(a, '/', detail::escape(b));
         });
-        return concat('(', str, ") ");
+
+        return concat('(', str, ") ") + get_byte_positions(leaf_element);
 #else
-        static_cast<void>(leaf_element);
-        return "";
+        return get_byte_positions(leaf_element);
 #endif
     }
 
   private:
     /// an exception object as storage for error messages
     std::runtime_error m;
+    template<typename BasicJsonType>
```

Comment:
The coverage report says this code is not covered:

<img width="1315" alt="image" src="https://github.com/user-attachments/assets/ba65b024-4e6d-42cb-bfca-83450f0741a6" />

Reply:
The coverage report doesn't build with `JSON_DIAGNOSTIC_POSITIONS` so there needs to be a file like `tests/src/unit-diagnostics.cpp` that always builds with it to get it into the coverage report.

Reply:
Yes. And for the combination with `JSON_DIAGNOSTICS`, see https://github.com/nlohmann/json/pull/4585#discussion_r1906530495.

Reply:
I added tests/src/unit-diagnostic-positions.cpp to cover this function.

Reply:
The case of `return ""` is still uncovered:

<img width="1160" alt="image" src="https://github.com/user-attachments/assets/7202c3e2-553e-4a3e-92f5-460da8bfb8e8" />

You need a test case where a JSON value that was not parsed, but user generated, created an error. Like

```json
json j = "foo";
int i = j.get<std::string>();
```

which should throw.

Reply:
![image](https://github.com/user-attachments/assets/849c443a-6b72-4870-b749-5f2c3c3e2d7d)
It looks like it is still not covered after I added a new test case!

Reply:
I first tried the below code snippet you suggested

```
json j = "foo";
int i = j.get<std::string>();
```
It was failing to compile as shown below

![image](https://github.com/user-attachments/assets/edc6bf34-0895-4b77-ba6d-179e187c7c44)

so I did this 
![image](https://github.com/user-attachments/assets/7b4c03f2-26a4-4a70-b130-b92f3e05f0b9)
it seems like the correct exception was thrown.

Reply:
See https://github.com/nlohmann/json/pull/4585#discussion_r1906060127 for why the coverage test isn't hitting it.